### PR TITLE
add entitlement routes

### PR DESCRIPTION
**Pull request by TroyKomodo**
## Proposed changes

Describe the big picture of what you want to change here, to help the maintainers understand why we should accept this pull request. Be sure to link to the issue if it fixes a bug or resolves a feature request.

## Types of changes

What types of changes does your code introduce to 7TV?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING](https://github.com/SevenTV/SevenTV/blob/main/CONTRIBUTING.md) doc
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, you may want to start the discussion by explaining why you chose the solution you did, what alternatives you considered, etc.

### CODE REVIEW

File: apps/data-brittler/src/jobs/special_events.rs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+				special_event::SpecialEvent::collection(global.target_db())
+					.untyped()
+					.create_indexes(indexes)
+					.await?;
+			}
+		}
+
+		Ok(Self { global })
+	}
+
```
Comment (lennartkloock) [inline apps/data-brittler/src/jobs/special_events.rs:36]: Are there any special events that we have to transfer with the brittler?

File: apps/data-brittler/src/jobs/subscriptions/benefits.rs [hunk 2]
```diff
@@ -0,0 +1,2168 @@
...
+		BenefitWithEntitlements::new(
+			"Subscriber Pack",
+			SubscriptionBenefitCondition::Duration(DurationUnit::Months(1)),
+			vec![
+				EntitlementEdgeKind::Badge {
+					badge_id: "62f97c05e46eb00e438a696a".parse().unwrap(),
+				},
+				EntitlementEdgeKind::Role {
+					role_id: "6076a86b09a4c63a38ebe801".parse().unwrap(),
+				},
```
Comment (lennartkloock) [inline apps/data-brittler/src/jobs/subscriptions/benefits.rs:983]: You only get the role after one month?

File: apps/api/src/http/egvault/redeem.rs [hunk 3]
```diff
@@ -70,6 +85,7 @@ pub struct RedeemResponse {
 	/// Url that the website will open
 	authorize_url: Option<String>,
 	/// list of ids of cosmetics that the user received
+	/// TODO: is this needed?
 	items: Vec<String>,
```
Comment (lennartkloock) [inline apps/api/src/http/egvault/redeem.rs:89]: No, we can remove it. The frontend doesn't need it.

**Comment by lennartkloock** [comment]
Does this also ignore subscription badges that have been granted too early?

File: apps/data-brittler/src/jobs/special_events.rs (hunk 1 above)
Comment (TroyKomodo) [reply apps/data-brittler/src/jobs/special_events.rs:36]: From the existing dataset no.

File: apps/data-brittler/src/jobs/subscriptions/benefits.rs (hunk 2 above)
Comment (TroyKomodo) [reply apps/data-brittler/src/jobs/subscriptions/benefits.rs:983]: In the calculation we will ceil

File: apps/data-brittler/src/jobs/special_events.rs (hunk 1 above)
Comment (lennartkloock) [reply apps/data-brittler/src/jobs/special_events.rs:36]: Why does this job exist then?

File: apps/data-brittler/src/jobs/subscriptions/benefits.rs (hunk 2 above)
Comment (lennartkloock) [reply apps/data-brittler/src/jobs/subscriptions/benefits.rs:983]: Wouldn't that mean you get the one month badge without being subbed for one month?

File: apps/data-brittler/src/jobs/subscriptions.rs [hunk 4]
```diff
@@ -28,6 +29,7 @@ pub struct SubscriptionsJob {
 	global: Arc<Global>,
 	subscriptions: HashMap<SubscriptionId, Subscription>,
 	periods: Vec<SubscriptionPeriod>,
+	edges: Vec<EntitlementEdge>,
```
Comment (TroyKomodo) [inline apps/data-brittler/src/jobs/subscriptions.rs:32]: should likely be a hashset

File: apps/data-brittler/src/jobs/special_events.rs (hunk 1 above)
Comment (TroyKomodo) [reply apps/data-brittler/src/jobs/special_events.rs:36]: It creates them

**Pull request merged**